benchmarks/benches/songs.rs

```diff
@@ -0,0 +1,211 @@
+mod datasets_paths;
+mod utils;
+
+use criterion::{criterion_group, criterion_main};
+use milli::update::Settings;
+use utils::Conf;
+
+fn base_conf(builder: &mut Settings) {
+    let displayed_fields = [
+        "id", "title", "album", "artist", "genre", "country", "released", "duration",
+    ]
+    .iter()
+    .map(|s| s.to_string())
+    .collect();
+    builder.set_displayed_fields(displayed_fields);
+
+    let searchable_fields = ["title", "album", "artist"]
+        .iter()
+        .map(|s| s.to_string())
+        .collect();
+    builder.set_searchable_fields(searchable_fields);
+
+    let faceted_fields = [
+        ("released-timestamp", "number"),
+        ("duration-float", "number"),
+        ("genre", "string"),
+        ("country", "string"),
+        ("artist", "string"),
+    ]
+    .iter()
+    .map(|(a, b)| (a.to_string(), b.to_string()))
+    .collect();
+    builder.set_faceted_fields(faceted_fields);
+}
+
+const BASE_CONF: Conf = Conf {
+    dataset: datasets_paths::SMOL_SONGS,
+    queries: &[
+        "john ",             // 9097
+        "david ",            // 4794
+        "charles ",          // 1957
+        "david bowie ",      // 1200
+        "michael jackson ",  // 600
+        "thelonious monk ",  // 303
+        "charles mingus ",   // 142
+        "marcus miller ",    // 60
+        "tamo ",             // 13
+        "Notstandskomitee ", // 4
+    ],
+    configure: base_conf,
+    primary_key: Some("id"),
+    ..Conf::BASE
+};
+
+fn bench_songs(c: &mut criterion::Criterion) {
+    let default_criterion: Vec<String> = milli::default_criteria()
+        .iter()
+        .map(|criteria| criteria.to_string())
+        .collect();
+    let default_criterion = default_criterion.iter().map(|s| s.as_str());
+    let asc_default: Vec<&str> = std::iter::once("asc(released-timestamp)")
+        .chain(default_criterion.clone())
+        .collect();
+    let desc_default: Vec<&str> = std::iter::once("desc(released-timestamp)")
+        .chain(default_criterion.clone())
+        .collect();
+
+    let basic_with_quote: Vec<String> = BASE_CONF
+        .queries
+        .iter()
+        .map(|s| {
+            s.trim()
+                .split(' ')
+                .map(|s| format!(r#""{}""#, s))
```

Comment:
Doesn't this just do the job?

```suggestion
                .map(|s| format!("{:?}", s))
```

Reply:
This works, but I don't think we should rely on the result of the `Debug` trait. Right now, it works (and it'll probably work forever), but since they wrote that in the std, I tend to never use it like that:
>>> Derived Debug formats are not stable, and so may change with future Rust versions. Additionally, Debug implementations of types provided by the standard library (libstd, libcore, liballoc, etc.) are not stable, and may also change with future Rust versions.

But we can use it, do as you want :grin:

Reply:
As you want but the real advantage of using the debug impl is that it also escape quotes and stuff like that :)
But, no we will keep it like that, don't change this line :)